src/Plugin/Field/VocabularyReferenceFieldItemListInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\open_vocabularies\Plugin\Field;
+
+use Drupal\Core\Field\FieldItemListInterface;
+
+/**
+ * Interface for vocabulary reference field item lists.
+ */
+interface VocabularyReferenceFieldItemListInterface extends FieldItemListInterface {
+
+  /**
+   * Filters out values of a specific vocabulary association.
+   *
+   * @param string $association_id
+   *   The target vocabulary association ID.
+   *
+   * @return $this
+   */
+  public function filterValuesByTargetAssociation(string $association_id): VocabularyReferenceFieldItemListInterface;
```

Comment:
Mmm, when I read `filterValuesByTargetAssociation` I expect to have the resulting list including the values by the passed association, not excluding them. Maybe `filterOutValuesByTargetAssociation`? 🤷‍♂️

Reply:
I know I have the same problem but I followed the core naming: `filterEmptyItems()`.

Reply:
Maybe `filterTargetAssociationItems()` ?

Reply:
Then it's more consistent at least with the filterEmptyItems :)